pallets/settlement/src/lib.rs

```diff
@@ -630,8 +637,8 @@ decl_module! {
         ///
         /// # Permissions
         /// * Portfolio
-        #[weight = <T as Config>::WeightInfo::reject_instruction()]
-        pub fn reject_instruction(origin, instruction_id: u64) {
+        #[weight = <T as Config>::WeightInfo::reject_instruction(*max_legs_count)]
+        pub fn reject_instruction(origin, instruction_id: u64, portfolio: PortfolioId, max_legs_count: u32) {
```

Comment:
Not being a Rust developer, or even a developer, this change confuses me slightly so this is likely a stupid question 😳(ignore me if it is as I know you're all busy).

`max_legs_count` has been added to `reject_instruction` similar to how it is already in `affirm_instruction` and `withdraw_affirmation`. (The description comments for these should possible be updated to define what `max_legs_count` does in each so I'm not as confused 😄).

In reject_instruction it appears to be used to limit the number of iterations for checking if the portfolio is in the legs. i.e. if it is set to 2 the iterator only checks sequentially if the any of the first two legs return a portfolio associated with a leg (or until the end of the legs vector if less than 2 legs exist). In the case of a leg vector larger than 2 the remaining legs do not get checked. 

I may be wrong but this seems unnecessary. Why would we not want to let it check the full leg vector? `.any` exits the iterator at the first true case so already ensure only as much of the legs are checked as needed.

Reply:
We need to ensure that `max_legs_count` is larger then the number of legs that match the passed in portfolio and matches at least 1 leg.

Reply:
Thanks for your response. I still don't understand why there is a need to count more than 1 leg with the portfolio or a limit on number of legs the portfolio is involved in. If the caller has custody of a portfolio for even a single leg of the instruction I would have expected their rejection should cause the rejection the entire settlement instruction. Hence why I thought `max_leg_count` was unnecessary in the function and the authorisation check should instead be. 

```
            ensure!(
                legs.iter().any(|(_, leg)| leg.from == portfolio || leg.to == portfolio),
                Error::<T>::UnauthorizedSigner
            );
```

Reply:
`max_legs_count` is needed for the transaction cost calculation.  We also need to verify that the passed in `max_legs_count` is valid.  Basically we need a safe way to calculate the transaction cost based on the input parameters.

Reply:
Thanks again for your response earlier. I still have a lot to learn. I now have an idea what it is needed for even if I don't fully understand. I would have thought rejecting the transaction for someone with a portfolio in only a single leg vs someone who has one in all legs should be the same as all legs are iterated through for subsequent `unsafe_unclaim_receipts` and `unchecked_release_locks`.  So I would have thought `max_leg_length` should be set as greater than or equal to `legs.len()` rather than a filtered subset.

I've one more annoying comment 😄 

The description for the `LegCountTooSmall` error is:
```
/// Legs count should matches with the total number of legs in which given portfolio act as `from_portfolio`.
```
Yet in your proposed edit you compare to ensure `leg_count <= max_legs_count`. I would have expected the error to read `LegCountTooHigh` or `MaxLegCountTooSmall`? 

Similarly in `fn filtered_legs` where this error is found it says `/// Also, ensures that the number of filtered legs is under the limit` by comparing `filtered_legs.len() as u32 <= max_filtered_legs,`. So I would expect the error to also read either `LegCountTooHigh` or `MaxLegCountTooSmall`.

Then a better error description might then be:
```
/// Legs count, in which given portfolio acts, should be less than or equal to the maximum leg count.
```
I removed "as `from_portfolio`" as for your suggestion it now applies to both `to` and `from` in the `reject_instruction` function.

Reply:
Let's document what these parameters do?